pkg/xds/dynconf/listener.go

```diff
@@ -105,6 +107,12 @@ func AddConfigRoute(proxy *core_xds.Proxy, rs *core_xds.ResourceSet, name string
 				},
 			},
 		)
+		// set the MaxDirectResponseBodySizeBytes value dynamically
+		if routeConfig.MaxDirectResponseBodySizeBytes == nil {
+			routeConfig.MaxDirectResponseBodySizeBytes = wrapperspb.UInt32(uint32(len(bytes)) + maxDirectResponseBodyPadding)
+		} else {
+			routeConfig.MaxDirectResponseBodySizeBytes = wrapperspb.UInt32(uint32(len(bytes)) + routeConfig.MaxDirectResponseBodySizeBytes.GetValue())
```

Comment:
This needs to be check but AFAIU this is per response so each endpoint accounts individually. So here you don't want to maxDirectReponse to be the sum of all. You instead want the size to be the max of all no?

```suggestion
			routeConfig.MaxDirectResponseBodySizeBytes = wrapperspb.UInt32(math.Max(uint32(len(bytes) + padding), routeConfig.MaxDirectResponseBodySizeBytes.GetValue())
```

Reply:
But we only call 1 endpoint at a time right? So the body is just 1 endpoint at once. Can't you just check this with starting an envoy and loading each config?